plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -550,12 +555,45 @@ private IcebergTableHandle tableHandleForSnapshot(
                 Optional.ofNullable(tableProperties.get(TableProperties.DEFAULT_NAME_MAPPING)),
                 table.location(),
                 table.properties(),
+                getTablePartitioning(session, table),
                 false,
                 Optional.empty(),
                 ImmutableSet.of(),
                 Optional.of(false));
     }
 
+    private Optional<IcebergTablePartitioning> getTablePartitioning(ConnectorSession session, Table icebergTable)
+    {
+        if (!isBucketExecutionEnabled(session) || icebergTable.specs().size() != 1) {
```

Comment:
Could you add a TODO about handling schema evolution related cases if that is something feasible to handle later on ?

Reply:
That would only be reasonable in very limited cases.  I don't think it is worrth a TODO